### PR TITLE
Add updated challenge style guide

### LINKED ISSUE #12993: Import challenge style guide into main repo
#### Issue Description

The [CurriculumExpansion](https://github.com/freeCodeCamp/CurriculumExpansion) repository has a [challenge style guide](https://github.com/freeCodeCamp/CurriculumExpansion/blob/master/challenge-style-guide.md) and I feel that this should be imported into the main repository (this one) and maintained and updated from here.

This will probably help in standardizing the challenges and address some of the concerns brought up in this issue's thread https://github.com/freeCodeCamp/freeCodeCamp/issues/9291. Additionally, there have been questions (speaking from [experience](https://github.com/freeCodeCamp/freeCodeCamp/pull/12950) e.g. should I use `<h4>Instructions</h4>` or `<hr>`) when trying to remember or think about how something should be formatted. This is probably very valuable now when we've been formatting and prepping the new challenges for prime time.

cc/ @HKuz @QuincyLarson

### PR DESCRIPTION
**Pull request by HKuz**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)

#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue (replace XXXX with an issue no): Closes #12993 

#### Description

Brings an updated version of the Challenge Style Guide from the Curriculum Expansion repository into fCC's main one. As suggested in issue #12993, the file is placed in the `./seed/` directory.

### CODE REVIEW

File: seed/challenge-style-guide.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+freeCodeCamp offers 1,200 hours of interactive coding challenges. These are 100% focused on the practical skill of building software. You code the entire time. You learn to code by coding.
+
+You can learn theory through free online university courses. freeCodeCamp will focus instead on helping you learn to code and practice by building apps.
+
+With that practical focus in mind, let’s talk about the requirements for our coding challenges. (Note that these requirements do not apply to our algorithm challenges, checkpoint challenges, or projects.)
+
+## The 2 minute rule
+
+Each challenge should be solvable within 120 seconds by a native English speaker who has completed the challenges leading up to it. This includes the amount of time it takes to read the directions, understand the seeded code, write their own code, and get all the tests to pass.
```
Comment (Greenheart) [inline seed/challenge-style-guide.md]:
```diff
-secondsby
+seconds by
```

**Review by Greenheart — CHANGES_REQUESTED** [review]
This is a great idea - it will help us know what to aim for when improving and reviewing challenges as well!

Apart from one small typo, I only have one other thought: Shouldn't we link to this document from the CONTRIBUTING.md document as well? Because that's the document I link people too, and this is useful information.

Keep it up! :smile: :+1:

File: seed/challenge-style-guide.md (hunk 1 above)
Comment (HKuz) [inline seed/challenge-style-guide.md]: @Greenheart - I can re-push my branch, but there's a space between "seconds" and "by" on my local version. Odd.

**Comment by HKuz** [comment]
@Greenheart yes, linking to this doc in CONTRIBUTING.md would be helpful - maybe we have it as a link in the "Other resources" section at the bottom?

File: seed/challenge-style-guide.md (hunk 1 above)
Comment (Greenheart) [inline seed/challenge-style-guide.md]: @HKuz Maybe you didn't `git add` your changes? I forget it all the time... :smiley:

**Comment by Greenheart** [comment]
@HKuz Yes - that would be great as most people only want to contribute fixes :blush:

**Comment by HKuz** [comment]
@Greenheart - I added this in the "Other resources" section, but I had to guess what the link would be (since this file isn't actually in `staging` or on GitHub yet). We'll need to check that it actually works once it's merged.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -418,6 +418,9 @@ Be sure to post in the PR conversation that you have made the requested changes.
 
 ### Other resources
 
+-   [Challenge Style Guide to create a new freeCodeCamp
```
Comment (Greenheart) [inline CONTRIBUTING.md]: Perhaps mention something about "or to keep in mind when updating existing ones." Or maybe that's too wordy or already implied by what's said already?

Comment (Greenheart) [inline CONTRIBUTING.md]: The link should work fine :blush:

Comment (HKuz) [inline CONTRIBUTING.md]: How about "Challenge Style Guide to create new or update existing freeCodeCamp challenges" ??

Comment (erictleung) [inline CONTRIBUTING.md]: Or just "Style Guide for freeCodeCamp Challenges", which I think implies new and existing challenges :smile:

Comment (Greenheart) [inline CONTRIBUTING.md]: @erictleung Haha awesome!

Comment (HKuz) [inline CONTRIBUTING.md]: Yes, simple is best 👍 - just made that fix.

**Review by Greenheart — APPROVED** [review]
LGTM!

Thanks @HKuz :smile:
